synapse/storage/events.py

```diff
@@ -2159,6 +2159,35 @@ def _delete_old_state_txn(self, txn, room_id, topological_ordering):
             ]
         )
 
+    @defer.inlineCallbacks
+    def is_event_after(self, event_id1, event_id2):
+        is_after = True
+
+        to_1, so_1 = yield self._get_event_ordering(event_id1)
+        to_2, so_2 = yield self._get_event_ordering(event_id2)
+
+        # Prevent updating if the existing marker is ahead in the stream
+        if to_1 > to_2:
+            is_after = False
+        elif to_1 == to_2 and so_1 >= so_2:
+            is_after = False
```

Comment:
Having travelled back in time to my boolean algebra (:heart:) days I have proof that your `>` should be `<`